build/testing/integration/api/api.go

```diff
@@ -777,6 +777,116 @@ func API(t *testing.T, ctx context.Context, client sdk.SDK, namespace string, au
 				require.Nil(t, result)
 			})
 		})
+
+		t.Run("Boolean", func(t *testing.T) {
+			// Have to use a fixed entity ID instead of UUID, because of percentage matching
+			t.Run("default match", func(t *testing.T) {
+				result, err := client.Evaluation().Boolean(ctx, &evaluation.EvaluationRequest{
+					NamespaceKey: namespace,
+					FlagKey:      "boolean_disabled",
+					EntityId:     "beover50",
+					Context: map[string]string{
+						"no": "match",
+					},
+				})
+
+				require.NoError(t, err)
+
+				assert.Equal(t, evaluation.EvaluationReason_DEFAULT_EVALUATION_REASON, result.Reason)
+				assert.False(t, result.Value, "value should be the flag state")
+			})
+
+			t.Run("percentage match", func(t *testing.T) {
+				result, err := client.Evaluation().Boolean(ctx, &evaluation.EvaluationRequest{
+					NamespaceKey: namespace,
+					FlagKey:      "boolean_disabled",
+					EntityId:     "fixed",
+					Context: map[string]string{
+						"no": "match",
+					},
+				})
+
+				require.NoError(t, err)
+
+				assert.Equal(t, evaluation.EvaluationReason_MATCH_EVALUATION_REASON, result.Reason)
+				assert.False(t, result.Value, "value should be threshold match value")
+			})
+
+			t.Run("segment match", func(t *testing.T) {
+				result, err := client.Evaluation().Boolean(ctx, &evaluation.EvaluationRequest{
+					NamespaceKey: namespace,
+					FlagKey:      "boolean_disabled",
+					EntityId:     "fixed",
+					Context: map[string]string{
+						"foo":  "baz",
+						"fizz": "nothing",
+					},
+				})
+
+				require.NoError(t, err)
+
+				assert.Equal(t, evaluation.EvaluationReason_MATCH_EVALUATION_REASON, result.Reason)
+				assert.True(t, result.Value, "value should be segment match value")
+			})
+		})
+
+		t.Run("Batch", func(t *testing.T) {
+			// Have to use a fixed entity ID instead of UUID, because of percentage matching
+			t.Run("batch evaluates variant, boolean, and not found", func(t *testing.T) {
+				result, err := client.Evaluation().Batch(ctx, &evaluation.BatchEvaluationRequest{
+					Requests: []*evaluation.EvaluationRequest{
+						{
+							NamespaceKey: namespace,
```

Comment:
This is outside the scope of this PR, but now that I see its usage and we have the opportunity to change the API.. wdyt about moving namespace key to top level instead of ineach inner request ? I think we briefly talked about this in the past.. but don't remember the conclusion. Perhaps we allow both but the inner namespace key can override the 'global' namespace?

Reply:
I think this is doable but requires us to create a number of different request types.
One type of when its called as a single requested and different nested type.

Seems doable though.

Or do we let folks just request across different namespaces?
I supposed that might be a footgun down the road w.r.t to authorization.

Reply:
@markphelps I guess I am more of a fan of explicitness. We wouldn't want users to perceive unexpected behavior when they use our API. We would have to call it out in the docs.

If a user has the same flag in two different namespaces, and they omit the namespace key in one of them, they might get an evaluation for a flag they would not want otherwise.

So in this case, I feel it would be better to just error instead of serving an unwanted evaluation. If that makes sense?

Reply:
@yquansah I think Mark was suggesting we drop namespacekey from the individual requests and put the namespace on the batch. So everything in the batch is scoped to a single namespace. Is that right @markphelps ?

I may of confused things with my question.

Reply:
@GeorgeMac yeah thats what i was suggesting. 

```
batchEval {
  namespace: "foo",
  requests: [],
}
```